blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/ShardStateManager.java

```diff
@@ -222,61 +235,56 @@ protected void updateSlotOnRead(SlotState slotState) {
             final int slot = slotState.getSlot();
             final long timestamp = slotState.getTimestamp();
             UpdateStamp.State state = slotState.getState();
+
+            //For slots in state "A", this would be last ingest time
+            //For slots in state "X", this would be last rollup time
             final long lastUpdateTimestamp = slotState.getLastUpdatedTimestamp();
 
-            UpdateStamp stamp = slotToUpdateStampMap.get(slot);
-            long nowMillis = new DateTime().getMillis();
-            if (stamp == null) {
```

Comment:
Can we put a comment how the "last rolled timestamp" and "last ingested timestamps" are being persisted to the UpdateStamp object in memory?  I think that would be enough for me to follow this when I look at it later.